# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/PrathamTumminakatti/ML1/blob/main/work/notebooks/w01_research_question.ipynb)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*


## My Lane: Predicting Declining Content Performance

I chose the lane of predicting declining content performance because the dataset contains historical SEO and engagement metrics for 30,000 content items from 32 pseudonymized clients. My research question is whether these metrics can be used to identify content that is likely to decline in performance. This is a useful problem because editors cannot manually review thousands of pages, and a decision-support system could help prioritize which content should be refreshed first.



## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

## The Question: Decision, Action, and Cost of a Wrong Call

This project aims to improve the decision of **which content pages should be reviewed or refreshed first**. The recommendations are intended for **SEO specialists and content editors**, who can use them to prioritize their work and focus on pages that are most likely to decline in performance. A wrong recommendation has two possible costs. A **false positive** may cause editors to spend time updating content that did not actually need attention, wasting time and resources. A **false negative** may fail to identify genuinely declining content, allowing it to continue losing search visibility and traffic before any action is taken.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [11]:
import pandas as pd
from google.colab import files
import os
print(os.getcwd())
print(os.listdir())
uploaded = files.upload()
df = pd.read_csv("content_refresh_anonymized.csv")
print("Dataset Shape:", df.shape)
print("Number of Clients:", df["client_id"].nunique())

print("\nTrend Distribution:")
trend_counts = df["trend_direction"].value_counts()
print(trend_counts)

print("\nContent Type Distribution:")
print(df["content_type"].value_counts())



/content
['.config', 'content_refresh_anonymized (1).csv', 'content_refresh_anonymized.csv', 'sample_data']


Saving content_refresh_anonymized.csv to content_refresh_anonymized (2).csv
Dataset Shape: (30000, 44)
Number of Clients: 32

Trend Distribution:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64

Content Type Distribution:
content_type
keyword article       27207
feedly article         2096
comparison article      697
Name: count, dtype: int64


## Quick Look at the Data

The starter dataset contains **30,000** content items collected from **32** pseudonymized clients. Among these items, **16,262 (54.2%)** have a **downward trend**, indicating that declining content is common enough to investigate. The dataset also includes multiple content types, with **27,207 keyword articles**, **2,096 Feedly articles**, and **697 comparison articles**, providing enough variation to study how different content characteristics relate to content decline.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

## Careful Words: What I Can and Can't Claim

This project will identify **observed patterns** and **directional relationships** between historical SEO and engagement metrics and declining content performance. The model is intended as a **decision-support** tool that helps SEO specialists and content editors prioritize which pages should be reviewed or refreshed first. It **cannot prove** that any feature causes content to decline, **cannot predict future Google algorithm updates or search rankings**, and **cannot guarantee** that refreshing a page will improve its performance. Any conclusions will be interpreted as associations rather than causal relationships.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.